src/proxy-worker.ts

```diff
@@ -96,6 +96,18 @@ class Tunnel extends nodeTunnel.Tunnel {
 										'device is not available on registered service instance',
 									);
 								}
+								const forwardSignature = `By=open-balena-vpn(${this.serviceId})`;
+								let forwardHeader: string[] = [];
+								if (req.headers.forwarded != null) {
```

Comment:
It probably makes more sense to just do:
```suggestion
								if (req.headers.forwarded != null) {
									if (req.headers.forwarded.includes(forwardSignature)) {
										throw new errors.HandledTunnelingError(
											'loop detected forwarding tunnel request',
										);
									}
									req.headers.forwarded = `${req.headers.forwarded},${forwardSignature}`
								} else {
									req.headers.forwarded = forwardSignature
								}
```
to avoid the need for creating so many arrays that need to be garbage collected